File: Milestone 6/benim-portfolio-springboot-api/src/main/java/gcu/edu/Benim/Portfolio/Controllers/SkillsController.java
package gcu.edu.Benim.Portfolio.Controllers;

import gcu.edu.Benim.Portfolio.Business.SkillsService;
import gcu.edu.Benim.Portfolio.Data.JPAModels.Skills;
import gcu.edu.Benim.Portfolio.Models.DTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

/**
 *  Controller class for Skills model to implement all the REST-ful API actions
 */
@RestController
public class SkillsController {
    @Autowired
    private SkillsService service;
    private DTO dto;

    // Non-Default Constructor
    public SkillsController(SkillsService service){
        this.service = service;
    }

    // API GET call for {application:host}/skills to pull all the skills stored in database
    // returns the DTO model with the result data and necessary server-end information
    @GetMapping("/skills")
    DTO allSkills(){
        dto = new DTO(200, "Returned All Skills!", this.service.getAllSkills());
        return dto;
    }

    // API POST call for {application:host}/skills to create a skills row in database
    // returns the DTO model with the result data and necessary server-end information
    @PostMapping("/skills")
    DTO newSkills(@RequestBody Skills newSkills){
        dto = new DTO(200, "Created Skills with skillsID = " + newSkills.getSkillsID(), this.service.createSkills(newSkills));
        return dto;
    }

    // API PUT call for {application:host}/skills/{id} to update the skills row in database with the given ID
    // returns the DTO model with the result data and necessary server-end information
    @PutMapping("/skills/{id}")
    DTO updatedSkills(@RequestBody Skills updatedSkills, @PathVariable Long id){
        dto = new DTO(200, "Updated Skills with ID = " + id , this.service.updateSkills(id, updatedSkills));
        return dto;
    }

    // API DELETE call for {application:host}/skills/{id} to delete the skills row in database with the given ID
    // returns the DTO model with the result data and necessary server-end information
    @DeleteMapping("/skills/{id}")
    DTO deleteSkills(@PathVariable Long id){
        this.service.deleteSkills(id);
        dto = new DTO(200, "Removed Skills with ID = " + id, null);
        return dto;
    }
}

File: benim-portfolio-express-api/controllers/skills.controller.js
const service = require("../services/skills.service");
const {DTO} = require("../models/DTO");


/*
*   Calls the skills service to return all skills
*   then returns the result in DTO model as JSON
*/
exports.getSkills = (req, res, next) => {
    service.getSkills(req, (error, results) => {
        if (error) {
            return res.status(400).send({ success: 0, data: "Bad request" });
        }
        let dto = DTO(200, "Success", results)
        res.json(dto);
    });
};

/*
*   Calls the skills service to create the given skills
*   then returns the result in DTO model as JSON
*/
exports.createSkills = (req, res, next) => {
    service.createSkills(req.body, (error, results) => {
        if (error) {
            return res.status(400).send({ success: 0, data: "Bad Request" + error });
        }
        let dto = DTO(200, "Success", results)
        res.json(dto);
    });
};

/*
*   Calls the skills service to update a row of skills with the given ID
*   then returns the result in DTO model as JSON
*/
exports.updateSkills = (req, res, next) => {
    service.updateSkills(req, req.body, (error, results) => {
        if (error) {
            return res.status(400).send({ success: 0, data: "Bad Request" + error });
        }
        let dto = DTO(200, "Success", results)
        res.json(dto);
    });
};

/*
*   Calls the skills service to delete a row of skills with the given ID
*   then returns the result in DTO model as JSON
*/
exports.deleteSkills = (req, res, next) => {
    service.deleteSkills(req, (error, results) => {
        if (error) {
            //dto.message = "Bad Request";
            return res.status(400).send({ success: 0, data: "Bad Request" + error });
        }
        let dto = DTO(200, "Success", results)
        res.json(dto);
    });
};

